modules/data.land/R/ic_process.R

```diff
@@ -119,9 +138,41 @@ ic_process <- function(settings, input, dir, overwrite = FALSE){
     if(input$output == "css"){
       settings$run$inputs[["pss"]][['path']]  <- gsub("css","pss", path_to_settings)
       settings$run$inputs[["site"]][['path']] <- gsub("css","site", path_to_settings)
+      
+      # IF: For now IC workflow is only working for ED and it's the only case for copying to remote
+      # but this copy to remote might need to go out of this if-block and change
+      
+      # Copy to remote and change paths if needed
+      if (settings$host$name != "localhost") {
+        
+        out.dir.remote   <- file.path(settings$host$folder, paste0(new.site$name, "_", input$source))
+        pss.file.remote  <- file.path(out.dir.remote, basename(settings$run$inputs[["pss"]][['path']]))
+        settings$run$inputs[["pss"]][['path']] <- pss.file.remote
+
+        css.file.remote  <- file.path(out.dir.remote, basename(settings$run$inputs[["css"]][['path']]))
+        settings$run$inputs[["css"]][['path']] <- css.file.remote
+        
+        site.file.remote <- file.path(out.dir.remote, basename(settings$run$inputs[["site"]][['path']]))
+        settings$run$inputs[["site"]][['path']] <- site.file.remote
+        
+        remote.execute.cmd(settings$host, "mkdir", c("-p", out.dir.remote))
+        PEcAn.utils::remote.copy.to(settings$host, pss.file.local, pss.file.remote)
+        PEcAn.utils::remote.copy.to(settings$host, css.file.local, css.file.remote)
+        PEcAn.utils::remote.copy.to(settings$host, site.file.local, site.file.remote)
+        
+        # update DB record
+        ic_input <- db.query(paste0("SELECT * from inputs where id = ", putveg.id), con)
+        fname    <- db.query(paste0("SELECT * from formats where id = ", ic_input$format_id), con)
+        mtype    <- db.query(paste0("SELECT type_string from mimetypes where id = ", fname$mimetype_id), con)
+        dbfile.input.insert(in.path = out.dir.remote, in.prefix = model_file[["file_name"]],
```

Comment:
(1) If you're adding the remote dbfile record to an existing Input record that was created during local processing, then this should be a dbfile.insert not a dbfile.input.insert, because the input file already exists and it is vital that it not be duplicated. That will also save you the trouble of not having to query the format or mime-type 
(2) This could (and probably should) be generalized to a function that copy **ANY** existing input folder from local to remote (while updating the DB records)? That would be a quite generally useful feature.